src/django_components/component.py

```diff
@@ -254,33 +258,76 @@ def get_template(self, context: Mapping) -> Template:
     ) -> str:
         # NOTE: This if/else is important to avoid nested Contexts,
         # See https://github.com/EmilStenstrom/django-components/issues/414
-        context = context_data if isinstance(context_data, Context) else Context(context_data)
-        prepare_context(context, component_id=self.component_id, outer_context=self.outer_context or Context())
+        context = context if isinstance(context, Context) else Context(context)
+        prepare_context(context, self.component_id)
         template = self.get_template(context)
 
-        # Associate the slots with this component for this context
-        # This allows us to look up component-specific slot fills.
-        def on_node(node: Node) -> None:
-            if isinstance(node, SlotNode):
-                trace_msg("ASSOC", "SLOT", node.name, node.node_id, component_id=self.component_id)
-                set_slot_component_association(context, node.node_id, self.component_id)
+        # Support passing slots explicitly to `render` method
+        if slots_data:
+            fill_content = self._fills_from_slots_data(slots_data, escape_slots_content)
+        else:
+            fill_content = self.fill_content
+
+        # If this is top-level component and it has no parent, use outer context instead
+        if not context[_PARENT_COMP_CONTEXT_KEY]:
+            context_data = self.outer_context.flatten()
+        if context_data is None:
+            context_data = {}
+
+        slots, resolved_fills = resolve_slots(
+            template,
+            component_name=self.registered_name,
+            context_data=context_data,
+            fill_content=fill_content,
+        )
 
-        walk_nodelist(template.nodelist, on_node)
+        # Available slot fills - this is internal to us
+        updated_slots = {
+            **context.get(_FILLED_SLOTS_CONTENT_CONTEXT_KEY, {}),
+            **resolved_fills,
+        }
 
-        if slots_data:
-            self._fill_slots(slots_data, escape_slots_content)
+        # For users, we expose boolean variables that they may check
+        # to see if given slot was filled, e.g.:
+        # `{% if variable > 8 and component_vars.is_filled.header %}`
+        slot_bools = {
+            slot_fill.escaped_name: slot_fill.is_filled for slot_fill in resolved_fills.values()
+        }
 
-        return render_component_template_with_slots(
-            self.component_id, template, context, self.fill_content, self.registered_name
-        )
+        with context.update(
+            {
+                _ROOT_CTX_CONTEXT_KEY: self.outer_context,
+                _FILLED_SLOTS_CONTENT_CONTEXT_KEY: updated_slots,
+                # NOTE: Public API for variables accessible from within a component's template
+                # See https://github.com/EmilStenstrom/django-components/issues/280#issuecomment-2081180940
+                "component_vars": {
+                    "is_filled": slot_bools,
```

Comment:
As noted above, this is the public API, and this where we get the `component_vars.is_filled` variable